### PR TITLE
migrate brn_icon_button.dart brn_vertical_icon_button.dart to null safety

### PR DESCRIPTION
**Pull request by cornflower10**

### CODE REVIEW

File: lib/src/components/button/brn_icon_button.dart [hunk 1]
```diff
@@ -82,13 +80,13 @@ class BrnIconButton extends StatefulWidget {
 class _BrnIconButtonState extends State<BrnIconButton> {
   @override
   Widget build(BuildContext context) {
-    Container ctn;
+    Container? ctn;
```
Comment (ShzMinato) [inline lib/src/components/button/brn_icon_button.dart:83]: 这个是不是 肯定不是null的嘞

Comment (cornflower10) [reply lib/src/components/button/brn_icon_button.dart:83]:
```
   if (widget.onTap != null) {
      return GestureDetector(
        child: ctn,
        onTap: () {
          widget.onTap!();
        },
      );
    }
    return ctn!;
  }
  ```
In the _BrnIconButtonState class, the GestureDetector child needs nullable parameters here.

File: lib/src/components/button/brn_icon_button.dart [hunk 2]
```diff
@@ -107,7 +105,7 @@ class _BrnIconButtonState extends State<BrnIconButton> {
                         fontSize: 11,
                         color: BrnThemeConfigurator.instance
                             .getConfig()
-                            .commonConfig
+                            .commonConfig!
```
Comment (zhoujuanjuan) [inline lib/src/components/button/brn_icon_button.dart:108]: already support not null, recommended to pull the latest code

File: lib/src/components/button/brn_icon_button.dart (hunk 1 above)
Comment (zhoujuanjuan) [reply lib/src/components/button/brn_icon_button.dart:83]: > ```
>  if (widget.onTap != null) {
>     return GestureDetector(
>       child: ctn,
>       onTap: () {
>         widget.onTap!();
>       },
>     );
>   }
>   return ctn!;
> }
> ```
> 
> In the _BrnIconButtonState class, the GestureDetector child needs nullable parameters here.

cool

**Comment by zhoujuanjuan** [comment]
感谢共建，因有重复提交，暂时关闭，可以参照https://github.com/LianjiaTech/bruno/issues/49 认领要迁移的项

**Pull request closed without merging**